### PR TITLE
Refactoring to async functions: utils.js and rules/html folder

### PR DESCRIPTION
**Pull request by hameleonka**
@rpl in this PR I refactored functions that return Promise to async functions (without await). There are 3 files that I've changed : utils.js and 2 files in rules/html folder. Please check it out, waiting for your review.

### CODE REVIEW

File: src/rules/html/warn-on-inline.js [hunk 1]
```diff
@@ -2,23 +2,21 @@ import { VALIDATION_WARNING } from 'const';
...
-            file: filename,
-          }));
-      }
-    });
-    resolve(linterMessages);
+export async function warnOnInline($, filename) {
+  const linterMessages = [];
+  $('script').each((i, element) => {
+    if ($(element).attr('src') === undefined &&
+      ($(element).attr('type') === undefined ||
```
Comment (rpl) [inline src/rules/html/warn-on-inline.js]: Nit, do you mind to align this line and the next line to the `$(...)` from the previous line?
e.g.
```
if ($(element)....
    ($(element)...
     $(element)...)) {
  ...
}
```

File: src/rules/html/warn-on-inline.js [hunk 2]
```diff
@@ -2,23 +2,21 @@ import { VALIDATION_WARNING } from 'const';
...
+        Object.assign({}, messages.INLINE_SCRIPT, {
+          /* This could occur in any HTML file, so let's make it
+            * a warning in case they've included any other file.
+            */
+          type: VALIDATION_WARNING,
+          file: filename,
+        }));
+    }
   });
+  return linterMessages;
```
Comment (rpl) [inline src/rules/html/warn-on-inline.js]: Nit (optional, but nice) add an empty line between the `});` and the `return linterMessage`.

Comment (hameleonka) [reply src/rules/html/warn-on-inline.js]: @rpl all changes have been made

File: src/rules/html/warn-on-inline.js (hunk 1 above)
Comment (hameleonka) [reply src/rules/html/warn-on-inline.js]: Done

**Pull request merged**